cfme/containers/image.py

```diff
@@ -122,7 +126,33 @@ def compliant(self):
             raise ValueError("{} is not a known state for compliance".format(text))
 
 
+@attr.s
+class ImageCollection(BaseCollection):
+    """Collection object for :py:class:`Image`."""
+
+    ENTITY = Image
+
+    def all(self):
+
+        image_table = self.appliance.db.client['container_images']
+        ems_table = self.appliance.db.client['ext_management_systems']
+        image_query = self.appliance.db.client.session.query(
+            image_table.name, image_table.image_ref, ems_table.name).join(
+            ems_table, image_table.ems_id == ems_table.id)
+        images = []
+
+        for name, id, provider_name in image_query.all():
+            images.append(self.instantiate(name=name, id=id,
+                provider=ContainersProvider(name=provider_name, appliance=self.appliance)))
+        return images
+
+    def get_random_instances(self, count=1):
```

Comment:
**Required**

If needed, this should be a test fixture, and not a convenience method on the collection class.

Please move this into the test modules.

Reply:
There was a test fixture on test modules that call this method.
According to your suggestion i move this implementation to test fixture into  test modules.